awscli/customizations/s3/utils.py

```diff
@@ -638,6 +642,15 @@ def _permission_to_param(cls, permission):
                          'read|readacl|writeacl|full')
 
     @classmethod
+    def _set_metadata_params(cls, request_params, cli_params):
+        if cli_params.get('metadata'):
+            request_params['Metadata'] = cli_params['metadata']
+
+    @classmethod
+    def _auto_populate_metadata_directive(cls, request_params):
+        request_params['MetadataDirective'] = 'REPLACE'
```

Comment:
Also only autopopulate it if ``Metadata`` is present in ``request_params``. I think that is why the tests are failing.